backend/LexBoxApi/Services/SendReceiveService.cs

```diff
@@ -0,0 +1,89 @@
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using LexBoxApi.Config;
+// using LexCore.Entities;
+using LexData;
+// using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
+using SIL.Progress;
+
+namespace LexBoxApi.Services;
+
+public class SendReceiveService
+{
+    private readonly LexBoxDbContext _dbContext;
+    private readonly IHgService _hgService;
+    private readonly IOptions<HgConfig> _hgOptions;
+    private readonly IOptions<SendReceiveConfig> _sendReceiveOptions;
+    private readonly IProgress _progress;
+
+    public SendReceiveService(LexBoxDbContext dbContext, IHgService hgService, IOptions<HgConfig> hgOptions, IOptions<SendReceiveConfig> sendReceiveOptions, IProgress progress)
+    {
+        _dbContext = dbContext;
+        _hgService = hgService;
+        _hgOptions = hgOptions;
+        _sendReceiveOptions = sendReceiveOptions;
+        _progress = progress;
+    }
+
+    public async Task<string> VerifyHgVersion()
+    {
+        string output = "";
+        using (Process hg = new()) {
+            string hgFilename = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? "hg.exe" : "hg";
+            hg.StartInfo.FileName = System.IO.Path.Join("Mercurial", hgFilename);
+            hg.StartInfo.Arguments = "version";
+            hg.StartInfo.RedirectStandardOutput = true;
+
+            hg.Start();
+            output = hg.StandardOutput.ReadToEnd();
+            await hg.WaitForExitAsync();
+        }
+        return output;
+    }
+
+    public async Task<string> CloneProject(string projectCode, string destDir)
+    {
+        var repoUrl = $"{_hgOptions.Value.HgWebUrl}/hg/{projectCode}";
```

Comment:
we should probably extract this url out into a helper method to make it in case we want to refactor it.

Reply:
Right, if the LexBoxApi is just the Language Depot equivalent then it doesn't belong there. I was thinking "Language Forge equivalent" when I did this. I'll move it.

Reply:
Done in commit f8941e8.